Orange GUI: strings improvement

sstanovnik opened the pull request:
The pull request fixes a few issues with spelling, phrasing, typos and such, all inside the Orange GUI. This is some low-hanging fruit for polishing the application. 

Fixes and improvements:
- spelling (plurals etc)
- proper articles
- consIstenT cApitalIzation
- three-dots are now attached to the last word, as is the norm (Save As ... -> Save As...)
- consistent usage of active/passive verbs. 

There are some changes I'm not completely sure about (the ones where I changed more than two letters) and I'd appreciate a review of them, so the meaning stays what it should be.

kernc commented on Orange/canvas/application/canvasmain.py:
```diff
@@ -1067,7 +1067,7 @@ def ask_save_changes(self):
         title = document.scheme().title or "untitled"
         selected = message_question(
-            self.tr('Do you want to save the changes you made to workflow "%s"?')
+            self.tr('Do you want to save the changes you made to the workflow "%s"?')
```
Article would be appropriate if there weren't a specific workflow name (proper noun) provided.

https://www.google.com/search?q="save+the+changes+you+made+to"

kernc commented on Orange/widgets/classify/owadaboost.py:
```diff
@@ -28,12 +28,12 @@ class OWAdaBoostClassification(OWBaseLearner):
+        self.base_label = gui.label(box, self, "Base Estimator: " + self.base_estimator.name)
 
-        gui.spin(box, self, "n_estimators", 1, 100, label="Number of estimators",
+        gui.spin(box, self, "n_estimators", 1, 100, label="Number of Estimators",
```
I don't think I like Title Case everywhere. Perhaps for section titles; [definitely not for option labels](https://www.google.si/search?tbm=isch&q=libreoffice+calc+preferences&imgrc=Vf_5Y57gkjehSM%3A&cad=h)!

kernc commented on Orange/widgets/classify/owlogisticregression.py:
```diff
@@ -11,7 +11,7 @@
 class OWLogisticRegression(OWBaseLearner):
     name = "Logistic Regression"
-    description = "Logistic regression classification algorithm with " \
+    description = "The logistic regression classification algorithm with " \
```
The classifier names in these contexts are proper nouns, algorithm (entity) names. There's no ambiguity which specific "Logistic regression" the description here refers to — there is only one.

ajdapretnar replied:
+1

janezd commented:
Inconsistencies in capitalization annoys me, too. It's good that you opened this.

I suggest we first define the rules, though. Not everything should be capitalized. We already had them years ago, but I can't find that page. Different applications follow different rules and neither MS nor Apple are always consistent, not even within the same dialog. 
- **Yes**: menu choices, push buttons, titles.
- **Probably yes**: combo box options (both MS and Apple mostly capitalize them, but with title case, e.g. `by`, not `By`).
- **No**: labels (including checkboxes).

@BlazZupan?

As for articles, I don't know the rules and instead follow the intuition. I would side with @kernc.

sstanovnik commented:
Thanks for the feedback. I like the suggested capitalization rules and have implemented them almost everywhere, except where they seem out of place to me. These are long button labels such as "Browse documentation data sets" in the File widget, "Send on change" in Paint Data. Thoughts?

About the classifier descriptions: using no articles seems weird to me, but maybe that's just my opinion. How about instead of just repeating the component name, we add a concise description classifier's main idea? I think that'd add a bit more value to the description. 

Another thing I noticed is inconsistent naming of the auto commit/commit button. Here are a few examples: auto commit, send on change, auto apply, auto send is on, commit on change, commit on change is on. There is a default preference in `gui.py`, but it's not really frequently used. We should consolidate this terminology, right?

ajdapretnar commented:
> Another thing I noticed is inconsistent naming of the auto commit/commit button. Here are a few examples: auto commit, send on change, auto apply, auto send is on, commit on change, commit on change is on. There is a default preference in gui.py, but it's not really frequently used. We should consolidate this terminology, right?

+1

ajdapretnar commented on Orange/widgets/data/owimpute.py:
```diff
@@ -205,7 +205,7 @@ def __init__(self):
 
         bgroup.buttonClicked[int].connect(self.set_default_method)
-        box = group_box(self.tr("Individual attribute settings"),
+        box = group_box(self.tr("Individual Attribute settings"),
```
Would it not make more sense to write "Individual Attribute Settings" then, all capitalized?

sstanovnik replied:
Whoops, I'll fix that. Thanks.

sstanovnik commented:
I've introduced the Apple Human Interface Guidelines for capitalization and colons and applied it. Also, I've redone some learner descriptions to be more descriptive (please check this) and consolidated the commit button text. This is now only apply/send, depending on the context in which it is used. The default capitalization was also changed to reflect the new guidelines. 

All discussed issues have now been addressed and are awaiting final review.

kernc commented on Orange/widgets/classify/ownaivebayes.py:
```diff
@@ -8,7 +8,8 @@
 class OWNaiveBayes(OWBaseLearner):
     name = "Naive Bayes"
-    description = "Naive Bayesian classifier."
+    description = "A simple probabilistic classifier based on " \
```
A _fast and_ simple ...

This is nice.

janezd commented on Orange/widgets/data/owconcatenate.py:
```diff
@@ -54,7 +54,7 @@ class OWConcatenate(widget.OWWidget):
                    "Intersection of attributes in all tables")
 
-    id_roles = ("Class attribute", "Attribute", "Meta attribute")
+    id_roles = ("Class Attribute", "Attribute", "Meta Attribute")
```
`id_roles` appear in the combo and hence shouldn't be capitalized.

janezd commented on Orange/widgets/data/owdatasampler.py:
```diff
@@ -83,21 +83,21 @@ def f():
             addSpace=12)
 
-        gui.appendRadioButton(sampling, "Cross Validation:")
+        gui.appendRadioButton(sampling, "Cross Validation")
```
Also change capitalization here ("Validation" to "validation")

janezd commented on Orange/widgets/data/owfile.py:
```diff
@@ -176,7 +176,7 @@ def __init__(self):
         box = gui.hBox(self.controlArea)
         gui.button(
-            box, self, "Browse documentation data sets",
+            box, self, "Browse Documentation Data Sets",
```
What about breaking the rule here? This button is so long that title case looks weird.

kernc replied:
:+1:

janezd commented on Orange/widgets/data/owpaintdata.py:
```diff
@@ -905,18 +905,18 @@ def _init_ui(self):
         gui.rubber(self.controlArea)
         gui.auto_commit(self.left_side, self, "autocommit",
-                        "Send", "Send on change")
+                        "Send")
```
I prefer the old auto send message. Before, it said "Send on change", now it is "Auto Send Is On" -- this is the default message set by `gui.auto_commit` function. This "auto XXX is on" is too technical, so I almost always overrode it when I was introducing the `auto_commit` function.

I suggest though that we change the default in `gui.auto_commit` to `label.title() + "on change"`. Note that I wouldn't use capital letters; although it's technically a button, it's always disabled, and, more importantly, the text is not written as a "command" (e.g. "Apply", "Do This") but rather as a message.

kernc replied:
```py
label.title().capitalize() + ' on change'
```

How about _Apply_ as default instead of _Send_ or _Commit_ or overriding everywhere?

janezd commented on Orange/widgets/data/owpreprocess.py:
```diff
@@ -623,15 +623,15 @@ def __init__(self, parent=None, **kwargs):
-        self.__centercb.addItems(["No centering", "Center by mean",
-                                  "Center by median"])
+        self.__centercb.addItems(["No Centering", "Center by Mean",
+                                  "Center by Median"])
```
This is combo - no capitals.

janezd commented on Orange/widgets/data/owpreprocess.py:
```diff
@@ -623,15 +623,15 @@ def __init__(self, parent=None, **kwargs):
-        self.__scalecb.addItems(["No scaling", "Scale by std",
-                                 "Scale by span"])
+        self.__scalecb.addItems(["No scaling", "Scale by SD",
+                                 "Scale by Span"])
```
Span -> span

janezd commented on Orange/widgets/data/owpurgedomain.py:
```diff
@@ -94,8 +94,7 @@ def __init__(self):
 
-        gui.auto_commit(self.buttonsArea, self, "autoSend", "Send Data",
-                        checkbox_label="Send automatically",
+        gui.auto_commit(self.buttonsArea, self, "autoSend", "Apply",
```
Also here: instead of "Send automatically" we now have "Auto Send Is On", which is uglier.

janezd commented on Orange/widgets/evaluate/owconfusionmatrix.py:
```diff
@@ -72,17 +72,17 @@ class OWConfusionMatrix(widget.OWWidget):
-                  "Proportion of actual"]
+    quantities = ["Number of Instances",
+                  "Proportion of Predicted",
+                  "Proportion of Actual"]
```
These go into a combo.

janezd commented on Orange/widgets/evaluate/owrocanalysis.py:
```diff
@@ -329,8 +329,8 @@ def __init__(self):
-                     items=["Merge predictions from folds", "Mean TP rate",
-                            "Mean TP and FP at threshold", "Show individual curves"],
+                     items=["Merge Predictions from Folds", "Mean TP Rate",
+                            "Mean TP and FP at Threshold", "Show Individual Curves"],
```
Combo.

janezd commented on Orange/widgets/unsupervised/owdistancefile.py:
```diff
@@ -55,7 +55,7 @@ def __init__(self):
     def set_file_list(self):
         super().set_file_list()
-        self.file_combo.addItem("Browse documentation data sets...")
+        self.file_combo.addItem("Browse Documentation Data Sets...")
```
If we decide to break the rule in the File widget, we should do it here, too.

janezd commented on Orange/widgets/unsupervised/owdistances.py:
```diff
@@ -14,9 +14,9 @@
     ("Jaccard", distance.Jaccard),
     ("Spearman", distance.SpearmanR),
-    ("Spearman absolute", distance.SpearmanRAbsolute),
+    ("Spearman Absolute", distance.SpearmanRAbsolute),
```
Combo

janezd commented on Orange/widgets/unsupervised/owdistances.py:
```diff
@@ -14,9 +14,9 @@
+    ("Spearman Absolute", distance.SpearmanRAbsolute),
     ("Pearson", distance.PearsonR),
-    ("Pearson absolute", distance.PearsonRAbsolute),
+    ("Pearson Absolute", distance.PearsonRAbsolute),
```
Combo

janezd commented on Orange/widgets/unsupervised/owhierarchicalclustering.py:
```diff
@@ -740,7 +740,7 @@ class OWHierarchicalClustering(widget.OWWidget):
     AttributeRole, ClassRole, MetaRole = 0, 1, 2
 
-    cluster_roles = ["Attribute", "Class variable", "Meta variable"]
+    cluster_roles = ["Attribute", "Class Variable", "Meta Variable"]
```
Combo

janezd commented on Orange/widgets/unsupervised/owkmeans.py:
```diff
@@ -25,13 +25,13 @@ class OWKMeans(widget.OWWidget):
+                       ("Inter-cluster Distance",
                         lambda km: km.inter_cluster, True),
-                       ("Distance to centroids",
+                       ("Distance to Centroids",
```
Combo

janezd commented on Orange/widgets/unsupervised/owkmeans.py:
```diff
@@ -75,15 +75,15 @@ def __init__(self):
 
         layout.addWidget(
-            gui.appendRadioButton(bg, "Optimized", addToLayout=False), 2, 1)
+            gui.appendRadioButton(bg, "Optimized from", addToLayout=False), 2, 1)
```
Add colon since the element introduces a control

janezd commented on Orange/widgets/unsupervised/owkmeans.py:
```diff
@@ -75,15 +75,15 @@ def __init__(self):
             controlWidth=60, alignment=Qt.AlignRight,
             callback=self.update_from)
-        gui.widgetLabel(ftobox, "  To: ")
+        gui.widgetLabel(ftobox, "to")
```
Add colon.

janezd commented on Orange/widgets/unsupervised/owkmeans.py:
```diff
@@ -125,15 +125,14 @@ def __init__(self):
         box = gui.vBox(self.controlArea, "Output")
         gui.comboBox(box, self, "place_cluster_ids",
-                     label="Append cluster id as ", orientation=Qt.Horizontal,
+                     label="Append cluster ID as", orientation=Qt.Horizontal,
```
Add colon

janezd commented on Orange/widgets/unsupervised/owmds.py:
```diff
@@ -107,20 +107,20 @@ class OWMDS(widget.OWWidget):
+        ("Every 5 Steps", 5),
+        ("Every 10 Steps", 10),
+        ("Every 25 Steps", 25),
+        ("Every 50 Steps", 50),
```
Combo

janezd commented on Orange/widgets/unsupervised/owmds.py:
```diff
@@ -316,13 +316,13 @@ def button(action):
+            items=["Original Features Only",
+                   "Coordinates Only",
+                   "Coordinates as Features",
+                   "Coordinates as Meta Attributes"],
```
Combo

janezd commented on Orange/widgets/unsupervised/owmds.py:
```diff
@@ -413,10 +413,10 @@ def _clear(self):
+        self.colorvar_model[:] = ["Same Color"]
+        self.shapevar_model[:] = ["Same Shape"]
+        self.sizevar_model[:] = ["Same Size"]
+        self.labelvar_model[:] = ["No Labels"]
```
Combo(s)

janezd commented on Orange/widgets/unsupervised/owmds.py:
```diff
@@ -441,9 +441,9 @@ def update_controls(self):
-            self.colorvar_model[:] = ["Same color", attr]
+            self.labelvar_model[:] = ["No Labels", attr]
+            self.shapevar_model[:] = ["Same Shape", attr]
+            self.colorvar_model[:] = ["Same Color", attr]
```
Combo(s)

janezd commented on Orange/widgets/unsupervised/owmds.py:
```diff
@@ -456,16 +456,16 @@ def update_controls(self):
                                           cont_vars)
-            self.labelvar_model[:] = chain(["No labels"],
+            self.labelvar_model[:] = chain(["No Labels"],
                                            [self.labelvar_model.Separator],
```
Combo(s)

janezd commented on Orange/widgets/unsupervised/owmds.py:
```diff
@@ -456,16 +456,16 @@ def update_controls(self):
                                           [self.sizevar_model.Separator],
                                           cont_vars)
-            self.labelvar_model[:] = chain(["No labels"],
+            self.labelvar_model[:] = chain(["No Labels"],
```
Combo(s)

janezd commented on Orange/widgets/unsupervised/owmds.py:
```diff
@@ -1053,10 +1053,10 @@ def send_report(self):
+            ("Color", self.color_value != "Same Color" and self.color_value),
+            ("Shape", self.shape_value != "Same Shape" and self.shape_value),
+            ("Size", self.size_value != "Same Size" and self.size_value),
+            ("Labels", self.label_value != "No Labels" and self.label_value)))
```
This is a report text. I guess no reason for capitalization here.

janezd commented on Orange/widgets/visualize/owscattermap.py:
```diff
@@ -524,7 +524,7 @@ def __init__(self):
         gui.comboBox(box, self, "color_scale", label="Scale: ",
                      orientation=Qt.Horizontal,
-                     items=["Linear", "Square root", "Logarithmic"],
+                     items=["Linear", "Square Root", "Logarithmic"],
```
Combo

kernc commented on CONTRIBUTING.md:
```diff
@@ -25,12 +25,6 @@ reproducible) and a screenshot (if applicable).
-Human interface guidelines 
---------------------------
-
-Use title case for titles, push buttons, menu titles and menu options. Elsewhere, use sentence case.
```
It's ok to link Crapple's HIG. It would be even better if the link was made, literally at will:
```md
... please conform to [Crapple's HIG]. ...

[Crapple's HIG]: https://developer.apple.com/library/mac/documentation/UserExperience/Conceptual/OSXHIGuidelines/TerminologyWording.html
```
In either case, leave the above innocuous sentence be.

janezd commented:
This may have seemed like a low-hanging fruit, but it was a lot of work and it's important to clean this up.

This is the type of commit that is bound to cause conflicts at rebasing, so we should give it some priority. I went through it and have two basic comments.

One is about combo boxes. I wrote "probably yes" in my first comment, but the Apple's rules you referenced as well as Microsoft's recommend the sentence case (with MS breaking the rule multiple times in [the very page where it decribes them](https://msdn.microsoft.com/en-us/library/windows/desktop/dn742404(v=vs.85).aspx).) I don't have strong feelings about this, but lean towards sentence case. I marked all the occurrences I noticed; there are not many of them, actually, so it's not a huge change. @BlazZupan, your opinion?

The other is about the default auto commit message. As I wrote in one of the comments, the current `auto_label = "Auto " + label.title() + " is On"` is ugly and technical. I suggest we change it to `label.title() + " on change"` or `label.title() + " automatically"`. I prefer the former, less technical.

sstanovnik commented:
Apple's rules don't really explicitly deal with combo box capitalization (only implicitly with the second-to-last table entry in the guidelines, I guess). Even in basic OS X software (iTunes, Finder, Settings), both variants are present without any special reason for them. We'll have to discuss this, probably with some "this looks better" criteria. 

About the colons: I left them out as the labels form parts of sentences (bottom of the HIG). I'd only add one to "Append cluster ID as", because that case is handled in the HIG, but definitely not beginh "from" and "to". 

I agree that "Auto Send is On" is more technical than "Send Automatically". The latter is a nice variant and I'm for using it, with its "Apply Automatically" sibling.

kernc commented:
_Auto-apply on change_

Everything is automatic in computing.

BlazZupan commented:
I agree with sentence case for combo boxes

janezd commented:
@BlazZupan, we checked a few cases with @sstanovnik. Sometimes we favoured title case, sometimes sentences. It would seem that title case is better when the combo is imperative (*Remove All Crocodiles*), and sentence case is better when the combo is descriptive (*Alligators as meat variables*).

@sstanovnik, is it done and ready for checking/merging?

sstanovnik commented:
@janezd Yes, all issues have been addressed.

kernc commented on CONTRIBUTING.md:
```diff
@@ -72,6 +66,17 @@ Please ensure your commits pass code quality assurance by executing:
+combo box options where the item is imperative (e.g. Initialize with Method)
+and sentence case otherwise.
+
+[OS X Human Interface Guidelines]: (https://developer.apple.com/library/mac/documentation/UserExperience/Conceptual/OSXHIGuidelines/TerminologyWording.html).
```
In this format, the URL is unparenthesized (to make a link). Also no trailing comma.

sstanovnik replied:
Whoops. Thanks for noticing, fixed.

janezd commented:
@sstanovnik, thanks, this was a great job, not a low-hanging fruit.
